addons/event_sale/tests/test_event_sale.py

```diff
@@ -73,6 +73,39 @@ def setUpClass(cls):
             {'name': 'Test Account Tax Group', 'company_id': cls.env.company.id}
         )
 
+    @users('user_sales_salesman')
+    def test_adjusted_quantity_in_sale_order(self):
+        """ This test ensures that when reducing the quantity of tickets for an event, we will cancel the
+        registrations for those tickets too.
+        """
+        customer_so = self.customer_so.with_user(self.env.user)
+        ticket = self.event_0.event_ticket_ids[0]
+
+        customer_so.write({
+            'order_line': [
+                (0, 0, {
+                    'event_id': self.event_0.id,
+                    'event_ticket_id': ticket.id,
+                    'product_id': ticket.product_id.id,
+                    'product_uom_qty': 3,
+                    'price_unit': 10,
+                })
+            ]
+        })
+
+        editor = self.env['registration.editor'].with_context({'default_sale_order_id': customer_so.id}).create({})
+        editor.action_make_registration()
+
+        registration_to_cancel = self.event_0.registration_ids[0]
+        registration_to_cancel.action_cancel()
+
+        registrations = self.env['event.registration'].search([('sale_order_id', '=', customer_so.id)])
+
+        expected_states = [['cancel'], ['draft'], ['draft']]
+        actual_states = [reg.sorted('id').mapped('state') for reg in registrations]
```

Comment:
No I means to `registrations.sorted('id').mapped('state')` so you'd get the exact same result.
Keeping `expected_states` a simple list

No need to make it a double list ^^

Reply:
Thank you again @reth-odoo , not sure what I was thinking, done 👍🏽